application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -122,16 +127,34 @@ private static boolean shouldBeArchived(MessageChannel channel, Instant archiveA
     }
 
     private void handleArchiveFlow(ThreadChannel threadChannel, MessageEmbed embed) {
+
         Consumer<Throwable> handleFailure = error -> {
             if (error instanceof ErrorResponseException) {
-                logger.warn("Unknown error occurred during help thread auto archive routine",
+                logger.warn(
+                        "Unknown error occurred during help thread auto archive routine, archiving thread",
                         error);
+
+                threadChannel.getManager().setArchived(true).queue();
             }
         };
 
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithMention =
+                member -> threadChannel.sendMessage(member.get().getAsMention()).addEmbeds(embed);
+
+        Function<Result<Member>, RestAction<Message>> sendEmbedWithoutMention =
+                member -> threadChannel.sendMessageEmbeds(embed);
+
         threadChannel.getGuild()
             .retrieveMemberById(threadChannel.getOwnerIdLong())
-            .flatMap(author -> threadChannel.sendMessage(author.getAsMention()).addEmbeds(embed))
+            .mapToResult()
+            .flatMap(member -> {
+                if (member.isSuccess()) {
+                    return sendEmbedWithMention.apply(member);
+                }
+
+                logger.debug("Unable to mention user", member.getFailure());
+                return sendEmbedWithoutMention.apply(member);
+            })
             .flatMap(any -> threadChannel.getManager().setArchived(true))
             .queue(any -> {
             }, handleFailure);
```

Comment:
do an `onErrorMap` so we'll always only do 1 queue

Reply:
not sure if i understand wym, could you show an example? Thanks

Reply:
```java
        .flatMap(...)
        .onErrorMap(any -> {
            log...
            return threadChannel.archive();
        })
        .queue();
```
Something like  this

Reply:
(or onErrorFlatMap, if that results in another rest action)